externals/llvm-external-projects/torch-mlir-dialects/include/torch-mlir-dialects/Dialect/Tcp/IR/TcpBase.td

```diff
@@ -40,4 +45,19 @@ class Tcp_Op<string mnemonic, list<Trait> traits = []> :
     Op<Tcp_Dialect, mnemonic, traits> {
 }
 
+class Tcp_UnaryElementwiseOp<string mnemonic, list<Trait> traits = []> :
+    Tcp_Op<mnemonic, !listconcat(traits, [
+        Pure,
+        Elementwise,
+        SameOperandsAndResultShape,
+        SameOperandsAndResultElementType])> {
+}
+
+class Tcp_BinaryElementwiseOp<string mnemonic, list<Trait> traits = []> :
+    Tcp_Op<mnemonic, !listconcat(traits, [
+        NoMemoryEffect,
```

Comment:
Not sure why binary ops are `NoMemoryEffect` but unary ops are `Pure`.

Reply:
Binary ops have UB if the shapes don't match.  Unary ops never have UB.